### DISCUSSION TITLE
How to read and write variables by Browse Name?

### DISCUSSION THREAD

**Discussion by AlexMacabu** [FreeOpcUa/opcua-asyncio#1228] [Q&A] [answered]
Hello! I would like to know how could I read and write values to an OPC UA Server (PLC) using the Browse Name or even Display Name?

Thanks a lot!

**Comment by AndreasHeine** [comment, marked as answer]
you can only use nodeids for direct read!
the only thing you can do use the translate browsepath to nodeid service to get the nodeid and then read.

**Comment by AlexMacabu** [comment]
Oh, too bad... In my case, I got an S7-1200 PLC and the default NodeId is
an index number (NUMERIC), which makes hard to track and can change if I
add/remove variables from the server. So being able to read/write variables
by their names would be ideal for me.

Thanks for the reply, though!

Em qua, 22 de fev de 2023 19:34, Andreas Heine ***@***.***>
escreveu:

**Reply by AndreasHeine** [reply]
@AlexMacabu the comination of browsing the tree and reading attibutes from found nodes should solve that.
then you can create a own map/dict with names to nodeid!

**Reply by AlexMacabu** [reply]
Thanks Andreas! I was able to work around this by changing the default nodeID from NUMERIC to STRING using another software called SiOME.
The document explaining step-by-step can be found here: 
https://support.industry.siemens.com/cs/document/109793221/how-do-you-change-the-node-id-identifier-type-of-the-nodes-in-the-s7-1200-opc-ua-server-from-numeric-to-string-?dti=0&lc=en-US

Now I just need to figure out how to write into an array of user-created data types.
